playbooks/files/maas_telegraf_format.py

```diff
@@ -22,6 +23,52 @@
 BASE_PATH = '/etc/rackspace-monitoring-agent.conf.d'
 RUN_VENV = '/usr/lib/rackspace-monitoring-agent/plugins/run_plugin_in_venv.sh'
 
+RETURN_QUEUE = multiprocessing.Queue()
+
+
+class Runner:
+    def __init__(self):
+        self.queue = multiprocessing.JoinableQueue()
+        self.processes = [
+            multiprocessing.Process(target=self.run_details)
+            for _ in range(2)
+        ]
+        for p in self.processes:
+            p.start()
+
+    def put(self, item):
+        self.queue.put(item)
+
+    def run_details(self):
+        while True:
+            detail_args = self.queue.get()
+            if not detail_args:
+                break
+
+            detail_args.insert(0, RUN_VENV)
+            detail_args.insert(2, '--telegraf-output')
+            with open(os.devnull, 'w') as null:
+                process = subprocess.Popen(
+                    ' '.join(detail_args),
```

Comment:
This is happening because each plugin is executed out of a venv which would require changing the interpreter (activating the venv), then running each plugin within that interpreter, which I'd rather not have to deal with. In the future, this will all be replaced with monitorstack so it will all eventually be deleted.